### PR TITLE
Update b-jet energy regression

### PR DESCRIPTION
**Pull request by andreypz**
#### PR description:

This is to update b-jet energy regression with the latest weights. Corresponding issue #432 

#### PR validation:
This has been validated manually in CMSSW_10_2_16_UL, and then ported to this master branch.

### CODE REVIEW

**Comment by peruzzim** [comment]
please rebase on the current master-cmsswmaster to fix the conflict

**Comment by andreypz** [comment]
Is `master-cmsswmaster` supposed to work now, without additional cms-merge XXX stuff?
I get a problem with  NanoAODDQM.cc
```
ReBase/CMSSW_11_0_0_pre11/src/PhysicsTools/NanoAOD/plugins/NanoAODDQM.cc:47:8: error: virtual function 'virtual void NanoAODDQM::endRun(const edm::Run&, const edm::EventSetup&)' overriding final function
```

**Comment by andreypz** [comment]
Never mind my previous comment - I now try with `CMSSW_11_0_0_pre10` instead of `CMSSW_11_0_0_pre11` and I don't get that error anymore.

**Comment by andreypz** [comment]
Hmm, now there are problems with `CMSSW_11_0_0_pre10`... I think it's because #447 just got merged. So I need to go back to `CMSSW_11_0_0_pre11`? 
I'll pause for now, will continue tomorrow, not to get caught in interference.

**Comment by peruzzim** [comment]
Yes sorry, we are updating right now.

The DQM error should not present itself if one uses releases consistently, because this DQM functionality was removed and, at the same time, the NanoAOD package was modified not to use it.
Actually the error does not present itself in our tests.
It could appear if you use the NanoAOD package without properly merging it in _pre11 (with merge-topic), or on the contrary if you update to _pre11 without the proper commands (this would keep using stuff compiled for _pre10).

**Comment by andreypz** [comment]
@peruzzim 
I think this should work now, although I'm not 100% sure about all the modifiers. Could you check those are correctly used.
In fact, is there a simple way to check which modifier is used when I run a production job?
What's confusing is that when I run  cmsDriver command wih `--era Run2_2017,run2_nanoAOD_94XMiniAODv2` option, I don't understand where the modifier
```
from Configuration.Eras.Modifier_run2_jme_2017_cff import run2_jme_2017
```
is picked up? That's the one I think should be used in `jets_cff.py`

**Comment by peruzzim** [comment]
Please configure the eras as follows: default using 2018 (not only with `run2_nanoAOD_102Xv1`), then modify it to the 2016 or 2017 trainings only for modifiers `run2_jme_2016` or `run2_jme_2017`, respectively.
They are activated as a part of `Run2_2016` and `Run2_2017` modifier chains.

**Comment by andreypz** [comment]
Ok, done. Please take a look.

File: PhysicsTools/NanoAOD/python/jets_cff.py [hunk 1]
```diff
@@ -410,6 +419,15 @@
 run2_miniAOD_80XLegacy.toModify( fatJetTable.variables, n3b1 = None)
 run2_jme_2016.toModify( fatJetTable.variables, jetId = Var("userInt('tightId')*2+userInt('looseId')",int,doc="Jet ID flags bit1 is loose, bit2 is tight"))
 
+run2_jme_2016.toModify( bjetNN, weightFile = cms.FileInPath("PhysicsTools/NanoAOD/data/breg_training_2016.pb") )
+run2_jme_2016.toModify( bjetNN,outputFormulas = cms.vstring(["at(0)*0.31976690888404846+1.047176718711853","0.5*(at(2)-at(1))*0.31976690888404846"]))
+
+run2_jme_2017.toModify( bjetNN, weightFile = cms.FileInPath("PhysicsTools/NanoAOD/data/breg_training_2017.pb") )
+run2_jme_2017.toModify( bjetNN,outputFormulas = cms.vstring(["at(0)*0.28225210309028625+1.055067777633667","0.5*(at(2)-at(1))*0.28225210309028625"]))
+
+run2_nanoAOD_102Xv1.toModify( bjetNN, weightFile = cms.FileInPath("PhysicsTools/NanoAOD/data/breg_training_2018.pb") )
```
Comment (peruzzim) [inline PhysicsTools/NanoAOD/python/jets_cff.py]: these lines should be removed, right?

File: PhysicsTools/NanoAOD/python/jets_cff.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 from Configuration.Eras.Modifier_run2_jme_2016_cff import run2_jme_2016
 from Configuration.Eras.Modifier_run2_jme_2017_cff import run2_jme_2017
 from Configuration.Eras.Modifier_run2_miniAOD_80XLegacy_cff import run2_miniAOD_80XLegacy
+from Configuration.Eras.Modifier_run2_nanoAOD_102Xv1_cff import run2_nanoAOD_102Xv1
```
Comment (peruzzim) [inline PhysicsTools/NanoAOD/python/jets_cff.py]: please also remove this as it's not used

File: PhysicsTools/NanoAOD/python/jets_cff.py (hunk 1 above)
Comment (andreypz) [reply PhysicsTools/NanoAOD/python/jets_cff.py]: Ok, I suppose the 2018 modifier should be removed as it is default now, but 2017 have to stay.

**Pull request merged**